crates/types/src/nodes_config.rs

```diff
@@ -147,9 +147,9 @@ pub enum Role {
 pub struct NodesConfiguration {
     version: Version,
     cluster_name: String,
-    // a unique fingerprint for this cluster. Introduced in v1.3 for forward compatibility. Will be
-    // used to uniquely identify this cluster instance in future versions.
-    #[serde(default)]
+    // A unique fingerprint for this cluster. Introduced in v1.3 for forward compatibility. As of
+    // v1.6, we expect this cluster fingerprint to be present. Will be used to uniquely identify
+    // this cluster instance. The value is None if the nodes configuration is invalid.
     cluster_fingerprint: Option<ClusterFingerprint>,
```

Comment:
I'd suggest we keep `#[serde(default)]` to not fail deserialization is the field is not present, therefore, we can still show a reasonable error message if we expect it to be Some. Otherwise, this will fail at deserialization with some obscure error message.